be/src/io/cache/block_file_cache.h

```diff
@@ -186,6 +189,9 @@ class BlockFileCache {
     bool try_reserve(const UInt128Wrapper& hash, const CacheContext& context, size_t offset,
                      size_t size, std::lock_guard<std::mutex>& cache_lock);
 
+    // async evict when water mark is reached, unlike try_reserve, removal is done in background
+    void try_evict_in_advance(size_t size, std::lock_guard<std::mutex>& cache_lock);
```

Comment:
add detailed comment what is the water mark, what is the behavor/machanism of this function, what is the assumption